### ISSUE TITLE
[Bug]: app_abfallplus_de returns names with numbers instead of the types

### ISSUE THREAD

**Issue by OnSive** [mampfes/hacs_waste_collection_schedule#2197]
### I Have A Problem With:

A specific source

### What's Your Problem

https://community.home-assistant.io/t/waste-collection-schedule-framework/186492/542

The attribute of my sensor do only contain names with numbers.
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/55302242/353201a8-10a1-4449-ac07-89fa2adb1270)

### Source (if relevant)

app_abfallplus_de / [ZAK Abfall App](https://play.google.com/store/apps/details?id=de.k4systems.abfallappzak)

### Logs

```Shell
N/A
```

### Relevant Configuration

Sensor (you can use the following wizard: `custom_components/waste_collection_schedule/waste_collection_schedule/wizard/app_abfallplus_de.py`) 

```YAML
waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.abfallappzak
        city: Fischen
        bezirk: Ortsteile
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by specularius** [comment]
Same here with these settings

waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.lkemmendingen
        city: Biederbach
        strasse: Alle Straßen

**Comment by 5ila5** [comment]
That's quite a neat bug.
Did some initial digging:
The app does return the correct data, but the reverse engineered way returns this super weird data (I do not know who's names these are and why they are returned here).

Debugging this is pretty annoying, as it's often not quite clear what went wrong exactly. I already did some trial and error but without success. I will take a deeper look in the coming days.

**Comment by OnSive** [comment]
@5ila5 can you give me a hint where to start debugging? Couldn't find a starting point as first time user of ha and waste wcs.

Btw. I hope that this is not real data as there would be way to many <18 year old people in a waste collector database...

**Comment by 5ila5** [comment]
## Generally debugging sources: 

1. Cloning the reposetory
2. modifing the files (sources are placed in custom_components/waste_collection_schedule/waste_collection_schedule/source/ some sources have a corresponding Service script (app_abfallplus_de has one), which is called by the source script. it should be located at custom_components/waste_collection_schedule/waste_collection_schedule/service/ 
3. You can test the modified source using the test script [documented here](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md#test-the-new-source-file) 

## app_abfallplus_de

To compare the calls made by the source with the request of the real app, you need: an android (vm?) device where you can inspect the traffic made by the phone (I use [Http Toolkit](https://httptoolkit.com/) to inspect the traffic made by my android VM running inside Android Studio)

Some changes like the User-Angent and client ID were made with an IPhone (as far as I can tell). But I tried to change them (Andoird has 2 different User-Agents for different requests and client_id must be 5 separate sections instead of one long integer string) and it did not solve anything, so I think the error is probably somewhere else.

**Comment by OnSive** [comment]
@5ila5 thanks that's helpful, maybe i can find a hint to the bug. Got the dev env running 👍

**Comment by 5ila5** [comment]
I put together the requests that are made by the app.
https://gist.github.com/5ila5/205672118616e30a2eaccbc89db42df4
Running this returns the XML with the names instead of the bin types, the app gets another XML (with the correct bin types), I do not find the difference between these requests and those made by the app.

**Comment by HenryLoenwind** [comment]
BTW, I suggest you complain to your local council (that is responsible for this _public service_) about this data being gated behind a private company that actively tries to prevent citizens from accessing it.

**Comment by Tigra16v** [comment]
The same Problem with these settings:

waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
       args:
          app_id: de.k4systems.abfallinfocw
          city: Egenhausen
          strasse: Alle Straßen

two days ago everything was still going.

**Comment by Fireblade900rr** [comment]
Also the same for my city.

waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.abfallinfoapp
        city: Weilerswist

**Comment by coarsy** [comment]
And here the same too. Waste collection schedulde is not running anymore:

waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.zakb
        city: Heppenheim
        strasse: Schubertweg
        hnr: 6
        
![bug_waste](https://github.com/mampfes/hacs_waste_collection_schedule/assets/2741933/861c38ac-4b1c-4ced-80c7-dc6ee2a11ed8)

**Comment by MatzzeF** [comment]
@HenryLoenwind pretty sure that this is a issue with the HA Plugin and not the data provider. The apps from the waste companies runs pretty fine here and shows the right data.

**Comment by mngsps** [comment]
My sensors are all "unknown" since 29th of june...
`waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.abfallappbh
        city: Au
        strasse: Alle Straßen`

**Comment by roninf** [comment]
Same here for:

`    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.abfallappbh`

Looks to me like birthdays....

**Comment by ReneNulschDE** [comment]
@5ila5, @OnSive : I made a few checks and some changes in a [test branch](https://github.com/mampfes/hacs_waste_collection_schedule/compare/master...ReneNulschDE:hacs_waste_collection_schedule:fix_abfallplusapp_2). With this changes I get the data again at least **once** for the first two postings (and leipziglk of course) in this thread. But needs some more time to analyze. I need to change the user agent after each execution. The name+number comes up with IOS user-agent only. With android user-agent the api delivers the bins but no dates...

![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/33263735/d506ada8-64fd-40dc-a680-78707984888e)

**Comment by boun** [comment]
@ReneNulschDE in my region `de.k4systems.zawdw` I did not get strange names, but calendar was empty (sometimes a strange date for a single event). I changed the code before the `self.get_collections()` call to set `self._client` to an id, that I sniffed from the app and sure enough, the calendar was filled up again. So there seems to be a problem in the set-up phase. I tested your changes but they did not fix the problem.

**Comment by garionth** [comment]
> My sensors are all "unknown" since 29th of june...
> 
> ```
> waste_collection_schedule:
>     sources:
>     - name: app_abfallplus_de
>       args:
>         app_id: de.k4systems.abfallappbh
>         city: Au
>         strasse: Alle Straßen
> ```

In your case i would suggest to check the log.
I had the same (sensors going to unknown) a couple of days ago. In the logs i could see that my provider (also app_abfallplus_de) changed the required parameters for the config.
The log said "Alle Straßen" is not a valid street name. And then gave a list of all the streetnames of my city.

btw thanks who ever is responsible for putting that return into the log. Very helpfull.

It was like in your case

city: real_city_name
strasse: Alle Straßen (all streets)

and is now
city: real_city_name
strasse: real_street_name
hsnr: real_number

**Comment by mngsps** [comment]
> > My sensors are all "unknown" since 29th of june...
> > ```
> > waste_collection_schedule:
> >     sources:
> >     - name: app_abfallplus_de
> >       args:
> >         app_id: de.k4systems.abfallappbh
> >         city: Au
> >         strasse: Alle Straßen
> > ```
> 
> In your case i would suggest to check the log. I had the same (sensors going to unknown) a couple of days ago. In the logs i could see that my provider (also app_abfallplus_de) changed the required parameters for the config. The log said "Alle Straßen" is not a valid street name. And then gave a list of all the streetnames of my city.
> 
> btw thanks who ever is responsible for putting that return into the log. Very helpfull.
> 
> It was like in your case
> 
> city: real_city_name strasse: Alle Straßen (all streets)
> 
> and is now city: real_city_name strasse: real_street_name hsnr: real_number

Thank you for the input!
In their App its still "Alle Straßen" (all streets) in my case.
Furthermore, nothing noticable in the logs:

```
2024-06-30 23:07:14.640 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration waste_collection_schedule which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-06-30 23:07:22.509 INFO (MainThread) [homeassistant.setup] Setting up waste_collection_schedule
2024-06-30 23:07:32.575 WARNING (MainThread) [homeassistant.setup] Setup of waste_collection_schedule is taking over 10 seconds.
2024-06-30 23:07:44.093 INFO (MainThread) [homeassistant.setup] Setup of domain waste_collection_schedule took 21.58 seconds
2024-06-30 23:07:44.095 INFO (MainThread) [homeassistant.components.sensor] Setting up waste_collection_schedule.sensor
2024-06-30 23:07:44.116 INFO (MainThread) [homeassistant.components.sensor] Setting up waste_collection_schedule.sensor
2024-06-30 23:07:44.119 INFO (MainThread) [homeassistant.components.sensor] Setting up waste_collection_schedule.sensor
2024-06-30 23:07:44.131 INFO (MainThread) [homeassistant.components.sensor] Setting up waste_collection_schedule.sensor
2024-06-30 23:07:55.478 INFO (MainThread) [homeassistant.components.calendar] Setting up waste_collection_schedule.calendar
```

**Comment by AKGitH** [comment]
Yesterday I wrote an e-mail to support@abfallplus.de and explained the API problem getting person names back.
![grafik](https://github.com/mampfes/hacs_waste_collection_schedule/assets/89026445/a1a242af-63ac-4a98-98a0-f4fe082f5839)

But till now I got no feedback, but it looks that someone is playing arround with the data, today I get "unkown" back.
Have a look to the data update date/time:
![grafik](https://github.com/mampfes/hacs_waste_collection_schedule/assets/89026445/6894e66d-bd90-4ae6-afe2-a69ac836758f)

**Comment by Tigra16v** [comment]
Same for me. After I tried it with street and house number and then undid it, I no longer have any names with numbers, just "unknown".

**Comment by anderl78** [comment]
For me too.... I get only "unknown"

**Comment by Tigra16v** [comment]
I found the solution. At least in my case it works. I noticed that the sensor calendar.apps_by_abfall was present multiple times. calendar.apps_by_abfall_1 calendar.apps_by_abfall_2 and so on. After I deleted all but one and restarted home assistant everything works again.

**Comment by roninf** [comment]
My calendar is empty meanwhile without any entry...

**Comment by anderl78** [comment]
> Crazy, but after a restart for other reasons its working as expected, witout changes.
> 
> waste_collection_schedule: sources:
> 
> * name: app_abfallplus_de
>   args:
>   app_id: de.k4systems.abfallinfoapp
>   city: Weilerswist

Doesn't work for me... after reboot still "unknow"

**Comment by anderl78** [comment]
> I found the solution. At least in my case it works. I noticed that the sensor calendar.apps_by_abfall was present multiple times. calendar.apps_by_abfall_1 calendar.apps_by_abfall_2 and so on. After I deleted all but one and restarted home assistant everything works again.
> 
> (In Settings -> Devices and services -> Entities)

Thank you. I got the clones too. Deleted them too, followed by a reboot - but still the same problem : state of my sensors is still "unknown"

**Comment by mngsps** [comment]
> > I found the solution. At least in my case it works. I noticed that the sensor calendar.apps_by_abfall was present multiple times. calendar.apps_by_abfall_1 calendar.apps_by_abfall_2 and so on. After I deleted all but one and restarted home assistant everything works again.
> > (In Settings -> Devices and services -> Entities)
> 
> Thank you. I got the clones too. Deleted them too, followed by a reboot - but still the same problem : state of my sensors is still "unknown"

Same for me. No change after removing duplicates...

**Comment by anderl78** [comment]
Oh, I just tested the abfallplus-app on my phone - it works without any problem... I even changed street and Number to see if it really gets new data...changed back, got my data back. Even checked there website - works well, all data is present there.

**Comment by Tigra16v** [comment]
> > I found the solution. At least in my case it works. I noticed that the sensor calendar.apps_by_abfall was present multiple times. calendar.apps_by_abfall_1 calendar.apps_by_abfall_2 and so on. After I deleted all but one and restarted home assistant everything works again.
> > 
> > (In Settings -> Devices and services -> Entities)
> 
> Thank you. I got the clones too. Deleted them too, followed by a reboot - but still the same problem : state of my sensors is still "unknown"

The sensor that is left must not have a number at the end. Otherwise it must be renamed. It must be called calendar.apps_by_abfall.

**Comment by mngsps** [comment]
> The sensor that is left must not have a number at the end. Otherwise it must be renamed. It must be called calendar.apps_by_abfall.

I did exactly this. No change, still "unknown"

**Comment by Tigra16v** [comment]
> > The sensor that is left must not have a number at the end. Otherwise it must be renamed. It must be called calendar.apps_by_abfall.
> 
> I did exactly this. No change, still "unknown"

I'm sorry. It helped me. Then there are probably other reasons. I wish you continued luck

**Comment by ReneNulschDE** [comment]
I spend some more hours on this topic. [Here](https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815) is a code that can be checked. (The code is not final but a starting point.)

Changes:
- Custom request header structure for the first request
- Not more than one request per second

```
Testing source app_abfallplus_de ...
  found 84 entries for de.k4systems.abfallappnf Ahrenviöl alle Straßen
  found 26 entries for de.albagroup.app Braunschweig Hauptstraße 7A  
  found 70 entries for de.k4systems.bonnorange Auf dem Hügel
  found 139 entries for de.ucom.abfallavr Brühl Habichtstr. 4A
  found 45 entries for de.k4systems.abfallappwug Bergen hauptstr. 1
  found 78 entries for de.k4systems.abfallappcux Wurster Nordseeküste Aakweg Alle Hausnummern
  found 40 entries for de.abfallwecker Mutzschen, Am Lindigt 1
  found 48 entries for de.k4systems.leipziglk Brandis Brandis
  found 46 entries for de.k4systems.leipziglk Machern Machern
  found 64 entries for de.k4systems.lkgoettingen, Abfallwirtschaft Altkreis Göttingen,  Adelebsen, Alle Straßen
```
manual testing (abfallappzak - Fischen, lkemmendingen - Biederbach, abfallinfoapp - Weilerswist)
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/33263735/5169a47e-9e7d-4fb5-b64f-8df1d8abc3d6)

**Comment by Lackii** [comment]
> I spend some more hours on this topic. [Here](https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815) is a code that can be checked. (The code is not final but a starting point.)
> 
Thanks a lot, this fixes the issue also for me. Before i have had the strange entrys with names and since monday an empty schedule. Clearing the double calendar.apps_by_abfall entrys did not changed anything.

**Comment by AKGitH** [comment]
After HA restart, I got this errors:
![grafik](https://github.com/mampfes/hacs_waste_collection_schedule/assets/89026445/d4d371f0-ec76-45e3-838e-1a829ee2f456)

Moving back to the original file, no errors showed, but still get "unkown".

**Comment by mngsps** [comment]
> I spend some more hours on this topic. [Here](https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815) is a code that can be checked. (The code is not final but a starting point.)
> 
> Changes:
> 
> * Custom request header structure for the first request
> * Not more than one request per second

Solved the problem in my case as well! Thank you!!

**Comment by ReneNulschDE** [comment]
@AKGitH : Your screenshot does not show the error message. Please share the log entry and your config.

**Comment by AKGitH** [comment]
Hi "ReneNulschDE",
I have downloaded the py-file again and now it works.
Great Job!

For your information: I use:
-name: app_abfallplus_de
-app_id: de.ucom.abfallavr
![xxx](https://github.com/mampfes/hacs_waste_collection_schedule/assets/89026445/593af329-4ce9-40c6-9fe3-7565e37ec6b9)

Many thanks!

**Comment by anderl78** [comment]
Btw, in case you need the information, Im using: 
name: app_abfallplus_de
app_id: de.k4systems.zawdw

**Comment by lorddagobert** [comment]
> I spend some more hours on this topic. [Here](https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815) is a code that can be checked. (The code is not final but a starting point.)
> 
> Changes:
> 
> * Custom request header structure for the first request
> * Not more than one request per second
> 
> ```
> Testing source app_abfallplus_de ...
>   found 84 entries for de.k4systems.abfallappnf Ahrenviöl alle Straßen
>   found 26 entries for de.albagroup.app Braunschweig Hauptstraße 7A  
>   found 70 entries for de.k4systems.bonnorange Auf dem Hügel
>   found 139 entries for de.ucom.abfallavr Brühl Habichtstr. 4A
>   found 45 entries for de.k4systems.abfallappwug Bergen hauptstr. 1
>   found 78 entries for de.k4systems.abfallappcux Wurster Nordseeküste Aakweg Alle Hausnummern
>   found 40 entries for de.abfallwecker Mutzschen, Am Lindigt 1
>   found 48 entries for de.k4systems.leipziglk Brandis Brandis
>   found 46 entries for de.k4systems.leipziglk Machern Machern
>   found 64 entries for de.k4systems.lkgoettingen, Abfallwirtschaft Altkreis Göttingen,  Adelebsen, Alle Straßen
> ```
> 
> manual testing (abfallappzak - Fischen, lkemmendingen - Biederbach, abfallinfoapp - Weilerswist) ![image](https://private-user-images.githubusercontent.com/33263735/345393722-5169a47e-9e7d-4fb5-b64f-8df1d8abc3d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwMzc1MDcsIm5iZiI6MTcyMDAzNzIwNywicGF0aCI6Ii8zMzI2MzczNS8zNDUzOTM3MjItNTE2OWE0N2UtOWU3ZC00ZmI1LWI2NGYtOGRmMWQ4YWJjM2Q2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDIwMDY0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzN2U3MWYzZjc0MWIzODBhOTRlNzMyOTRiYjEwNWE1ZDUyNmI2NDMyODJmNDcyOWRkOGFkOGU0YTU3YjQ4MDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.33UAZ-euiw7i1PlTA1olW8g0c8FLbu5bNvJBsgr3qoc)

Solved the problem also in my case as well! Thank you!!

name: app_abfallplus_de
app_id: de.idcontor.abfalllu

Many thanks again

**Comment by tobibachert** [comment]
> I spend some more hours on this topic. [Here](https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815) is a code that can be checked. (The code is not final but a starting point.)
> 
> Changes:
> 
> * Custom request header structure for the first request
> * Not more than one request per second
> 
> ```
> Testing source app_abfallplus_de ...
>   found 84 entries for de.k4systems.abfallappnf Ahrenviöl alle Straßen
>   found 26 entries for de.albagroup.app Braunschweig Hauptstraße 7A  
>   found 70 entries for de.k4systems.bonnorange Auf dem Hügel
>   found 139 entries for de.ucom.abfallavr Brühl Habichtstr. 4A
>   found 45 entries for de.k4systems.abfallappwug Bergen hauptstr. 1
>   found 78 entries for de.k4systems.abfallappcux Wurster Nordseeküste Aakweg Alle Hausnummern
>   found 40 entries for de.abfallwecker Mutzschen, Am Lindigt 1
>   found 48 entries for de.k4systems.leipziglk Brandis Brandis
>   found 46 entries for de.k4systems.leipziglk Machern Machern
>   found 64 entries for de.k4systems.lkgoettingen, Abfallwirtschaft Altkreis Göttingen,  Adelebsen, Alle Straßen
> ```
> 
> manual testing (abfallappzak - Fischen, lkemmendingen - Biederbach, abfallinfoapp - Weilerswist) ![image](https://private-user-images.githubusercontent.com/33263735/345393722-5169a47e-9e7d-4fb5-b64f-8df1d8abc3d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNzU3MTIsIm5iZiI6MTcyMDA3NTQxMiwicGF0aCI6Ii8zMzI2MzczNS8zNDUzOTM3MjItNTE2OWE0N2UtOWU3ZC00ZmI1LWI2NGYtOGRmMWQ4YWJjM2Q2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA0VDA2NDMzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZTFkMTgwNmU3MTE0ZmVmN2Y0NzU1MzQwMDNlMWJiM2E1Y2Q4MDRmN2QzOWI2MGE5NjYyZDhmZTY4YTNhOTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Red3KGBIdcl4KS30BX3p5VPVy1Hn_9zSLlPOywiDPP0)

THAT solved it for me. 
Thanks a lot @ReneNulschDE

**Comment by Tigra16v** [comment]
After the last core update I have the problem again. (sensor status unknown) Since I'm still a complete beginner and my English isn't great, I honestly don't understand what exactly solved the problem for you. Can someone please explain to me what exactly I need to do?

**Comment by coarsy** [comment]
@Tigra16v You have to replace the content of the file AppAbfallplusDe.py with this: https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815

**Comment by Tigra16v** [comment]
ok. and how exactly do I do that? I can't copy the text in the link (in the github app).

**Comment by thoduh** [comment]
Press the Link, then Press the tree Points:
<img width="1587" alt="Bildschirmfoto 2024-07-04 um 17 03 58" src="https://github.com/mampfes/hacs_waste_collection_schedule/assets/22728076/71bdaad1-fd77-4cfb-8615-44f3f477f664">
Press "View File"
Press "Copy raw file"
Paste in Homeassistant...

**Comment by thoduh** [comment]
> @Tigra16v You have to replace the content of the file AppAbfallplusDe.py with this: [6051222](https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815)

worked. thank you!

**Comment by Tigra16v** [comment]
OK. I got it working. But via the browser and not in the app. But now nothing works. The sensors are no longer available and I get the following error message

Logger: homeassistant.setup
Quelle: setup.py:485
Erstmals aufgetreten: 17:22:14 (5 Vorkommnisse)
Zuletzt protokolliert: 17:22:14

Unable to prepare setup for platform 'waste_collection_schedule.sensor': Unable to set up component.

**Comment by chma73** [comment]
@Tigra16v I think your AppAbfallplusDe.py is not correct, are you sure that you did the copy and replace correct.

**Comment by chma73** [comment]
@ReneNulschDE thanks for your effort and support, your update works fine for me

**Comment by Tigra16v** [comment]
At least I'm not aware of what I did wrong. I also get the following error message:Logger: homeassistant.setup
Quelle: setup.py:404
Erstmals aufgetreten: 18:25:35 (1 Vorkommnisse)
Zuletzt protokolliert: 18:25:35

Error during setup of component waste_collection_schedule
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 404, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/__init__.py", line 113, in async_setup
    await hass.async_add_executor_job(
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/__init__.py", line 201, in add_source_shell
    new_shell = SourceShell.create(
                ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 204, in create
    source_module = importlib.import_module(
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/app_abfallplus_de.py", line 1, in <module>
    import waste_collection_schedule.service.AppAbfallplusDe as AppAbfallplusDe
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py", line 541
    for a in soup.find_all(
                          ^
SyntaxError: '(' was never closed

**Comment by chma73** [comment]
@Tigra16v bist du noch in einem anderen Forum aktiv, bei dem man dir direkt schreiben kann?

**Comment by Tigra16v** [comment]
> @Tigra16v bist du noch in einem anderen Forum aktiv, bei dem man dir direkt schreiben kann?

Leider nicht. Aber bin offen für Vorschläge

**Comment by chma73** [comment]
@Tigra16v könnte das hier empfehlen https://community.simon42.com, wenn du den gleichen nick verwendest gib mir einfach bescheid..

**Comment by ReneNulschDE** [comment]
@Tigra16v : I strongly recommend to wait until the code has been merged into the official repo, especially if you have not yet gained so much experience. (My code example was just for testing for "experienced users" and to validate my work).  

If you can't/won't wait, click on [this link](https://raw.githubusercontent.com/mampfes/hacs_waste_collection_schedule/6051222586494d6e31c1e1bf37542e91dc780815/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py) and copy the complete content into the `service/AppAbfallplusDe.py`. Then restart HA.

Translated with DeepL.com (free version)

@Tigra16v : Ich empfehle dringend zu warten bis der Code in das offizielle Repo eingeflossen ist, speziell wenn Du noch nicht so viel Erfahrung gesammelt hast. (Mein Code Beispiel war nur zum Testen für "erfahrene User" und zur Validierung meiner Arbeit.)  

Wenn Du nicht warten kannst/willst, klicke auf [diesen Link](https://raw.githubusercontent.com/mampfes/hacs_waste_collection_schedule/6051222586494d6e31c1e1bf37542e91dc780815/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py) und kopiere den kompletten Inhalt in die `service/AppAbfallplusDe.py`. Starte danach HA neu.

**Comment by Tigra16v** [comment]
It's done. I've got it working now. I copied the file into the directory again using the command line. Now everything works. Great 👍 Thank you for the help!

**Comment by AndiHaas85** [comment]
@ReneNulschDE thanks for your work

tried your AppAbfallplusDe.py and it worked for me too

**Comment by thoduh** [comment]
de.k4systems.abfallapploe
Unfortunately, it only works partially:
<img width="453" alt="Bildschirmfoto 2024-07-05 um 20 01 35" src="https://github.com/mampfes/hacs_waste_collection_schedule/assets/22728076/60198635-de59-4703-8580-9e0570a9bd32">

**Comment by chrizzo84** [comment]
> I spend some more hours on this topic. [Here](https://github.com/mampfes/hacs_waste_collection_schedule/commit/6051222586494d6e31c1e1bf37542e91dc780815) is a code that can be checked. (The code is not final but a starting point.)
> 
> Changes:
> 
> * Custom request header structure for the first request
> * Not more than one request per second
> 
> ```
> Testing source app_abfallplus_de ...
>   found 84 entries for de.k4systems.abfallappnf Ahrenviöl alle Straßen
>   found 26 entries for de.albagroup.app Braunschweig Hauptstraße 7A  
>   found 70 entries for de.k4systems.bonnorange Auf dem Hügel
>   found 139 entries for de.ucom.abfallavr Brühl Habichtstr. 4A
>   found 45 entries for de.k4systems.abfallappwug Bergen hauptstr. 1
>   found 78 entries for de.k4systems.abfallappcux Wurster Nordseeküste Aakweg Alle Hausnummern
>   found 40 entries for de.abfallwecker Mutzschen, Am Lindigt 1
>   found 48 entries for de.k4systems.leipziglk Brandis Brandis
>   found 46 entries for de.k4systems.leipziglk Machern Machern
>   found 64 entries for de.k4systems.lkgoettingen, Abfallwirtschaft Altkreis Göttingen,  Adelebsen, Alle Straßen
> ```
> 
> manual testing (abfallappzak - Fischen, lkemmendingen - Biederbach, abfallinfoapp - Weilerswist) ![image](https://private-user-images.githubusercontent.com/33263735/345393722-5169a47e-9e7d-4fb5-b64f-8df1d8abc3d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAyMzg3ODMsIm5iZiI6MTcyMDIzODQ4MywicGF0aCI6Ii8zMzI2MzczNS8zNDUzOTM3MjItNTE2OWE0N2UtOWU3ZC00ZmI1LWI2NGYtOGRmMWQ4YWJjM2Q2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA2VDA0MDEyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1MjRlMDAzZGQ5MTk4YjdjYTNhZTQ4NTY0NTczMjg0NTFhN2NmY2NkOWUxZGQ0MmM3MDQ5NGMzMmJkZTFhMDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QSf4hi8YUv1tp9SZcAcrXnX1m6yjNJoX8TW0UymuPfo)

Thx a lot!! Also working for me :)

**Comment by specularius** [comment]
tried the modified AppAbfallplusDe.py. And bingo it works for me...

**Comment by ronnyadam** [comment]
the modified AppAbfallplusDe.py also works for me, thx!

`waste_collection_schedule:
  sources:
    - name: app_abfallplus_de
      args:
        app_id: de.k4systems.lkmabfallplus
        city: Feldkirchen`

**Comment by anderl78** [comment]
Hmm, is it just me, but the integration stopped working again here... the sensores became"unknown" since yesterday during the afternoon

**Comment by anderl78** [comment]
It's working again... I have changed the address, then did a HA restart = Data for the changed address (only one street away from my home) are there. Then changed back to my data, reboot and the data is back...

**Comment by dagobert** [comment]
I also had this problem with meinawblm
With the mentioned commit This is fixed.

When will this be incorporated in the normal release?

**Comment by 5ila5** [comment]
I'd probably release the next version once the fix is merged. @ReneNulschDE's fix seems to work, but I'll wait for him to ultimately decide if his fix is ready and for him to open a PR.

**Comment by ReneNulschDE** [comment]
@5ila5 : I have the PR ready more or less. Main problem is that with the current master branch the yaml-setup does not work. Config flow setup works fine but all my tests are depending on yaml setup. Let me know if you need a new issue or if this is known already.

**Comment by 5ila5** [comment]
This fix is now part of the 2.0.0 release

**Comment by dagobert** [comment]
> This fix is now part of the 2.0.0 release

May I ask about your release schedule and until when I need to stop updating?
Thanks!

**Comment by 5ila5** [comment]
@dagobert I generally publish a release around every month (sometimes sooner, sometimes later). I really don't understand your second question, the 2.0.0 release is released and HACS should prompt you to update. Updates typically include fixed source (mostly because the service provider changed their API/website), added sources and sometimes core feature updates (like GUI configuration this time)
